(#1082) Replace forbidden API calls

Umbrah opened the pull request:
This is for #1082 
I replaced forbidden API calls in org.cactoos.collection package, updated to-do.

0crat commented:
Job [#1114](https://github.com/yegor256/cactoos/issues/1114) is now in scope, role is `REV`

<!-- https://www.0crat.com/footprint/C63314D6Z/e47a7ec6-69f3-4b90-9708-e4fc818b7dd7, version: 0.46.6, hash: ${buildNumber} -->

0crat commented:
This pull request [#1114](https://github.com/yegor256/cactoos/issues/1114) is assigned to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), here is [why](https://www.0crat.com/footprint/C63314D6Z/8160943f-0713-452e-97e9-fbb23ebdc0f0); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

<!-- https://www.0crat.com/footprint/C63314D6Z/741d1a92-6346-40ef-a1f5-abd48d556eb6, version: 0.46.6, hash: ${buildNumber} -->

ilyakharlamov commented on src/test/java/org/cactoos/collection/BehavesAsCollection.java:
```diff
@@ -62,26 +62,52 @@ public BehavesAsCollection(final E item) {
     public boolean matchesSafely(final Collection<E> col) {
         MatcherAssert.assertThat(
-            col, new IsCollectionContaining<>(new IsEqual<>(this.sample))
+            "Must contain item",
```
Even though `MatcherAssert.assertThat` is still used, we are in transition to OO/Junit5 `new Assertion` for all the new code added.
Please use `new Assertion<>().affirm()`

ilyakharlamov commented on src/test/java/org/cactoos/collection/FilteredTest.java:
```diff
@@ -54,6 +54,7 @@ public void behavesAsCollection() throws Exception {
     @Test
     public void filterList() {
         MatcherAssert.assertThat(
+            "Length must change",
```
`new Assertion<>().affirm()`

llorllale replied:
@umbrah message should be "length equal to number of items matching filter"

llorllale replied:
@umbrah actually I don't understand what the difference is between this test and the `size()` test below. I think this one can be deleted

ilyakharlamov commented on src/test/java/org/cactoos/collection/StickyTest.java:
```diff
@@ -79,6 +83,7 @@ public void decoratesArray() throws Exception {
     @Test
     public void testEmpty() {
         MatcherAssert.assertThat(
+            "Must be empty",
```
`new Assertion<>().affirm()`

ilyakharlamov commented on src/test/java/org/cactoos/collection/TailOfTest.java:
```diff
@@ -44,9 +48,13 @@ public void tailCollection() throws Exception {
-            Matchers.contains(
-                "three",
-                "four"
+            new IsCollectionContaining<>(
```
`throws Exception` is not needed here

Umbrah commented:
@ilyakharlamov I replaced all MatcherAssert.assertThat with new Assertion<>().affirm() and removed throws Exception where it is not needed, please check

ilyakharlamov commented:
@llorllale it's good to merge

llorllale commented on src/test/java/org/cactoos/collection/HeadOfTest.java:
```diff
@@ -38,18 +42,22 @@
-                "two",
-                "three"
+            new IsCollectionContaining<>(
+                new AnyOf<>(
```
@Umbrah are you sure the logic is preserved here? Why not use `AllOf`?

llorllale commented on src/test/java/org/cactoos/collection/BehavesAsCollection.java:
```diff
@@ -61,28 +62,54 @@ public BehavesAsCollection(final E item) {
+        new Assertion<>(
+            "Must contain item",
+            () -> col,
+            new IsCollectionContaining<>(
```
@umbrah can't you use `HasValues`?

llorllale commented on src/test/java/org/cactoos/collection/BehavesAsCollection.java:
```diff
@@ -61,28 +62,54 @@ public BehavesAsCollection(final E item) {
+        new Assertion<>(
+            "Size must be more than 0",
+            () -> col,
+            new IsCollectionWithSize<>(
```
@umbrah `HasSize`

llorllale commented on src/test/java/org/cactoos/collection/BehavesAsCollection.java:
```diff
@@ -61,28 +62,54 @@ public BehavesAsCollection(final E item) {
+            () -> new ListOf<>(
+                (E[]) col.toArray()
+            ),
+            new IsCollectionContaining<>(
```
@umbrah `HasValues`

llorllale commented on src/test/java/org/cactoos/collection/BehavesAsCollection.java:
```diff
@@ -61,28 +62,54 @@ public BehavesAsCollection(final E item) {
+        new Assertion<>(
+            "Array from collection must contain item",
+            () -> new ListOf<>(array),
+            new IsCollectionContaining<>(
```
@umbrah `HasValues`

llorllale commented on src/test/java/org/cactoos/collection/FilteredTest.java:
```diff
@@ -43,68 +43,73 @@
-            new Filtered<String>(
+    public void size() {
+        new Assertion<>(
+            "Size must change",
```
@umbrah message must be "size is equal to number of items matching the filter"

llorllale commented on src/test/java/org/cactoos/collection/ReversedTest.java:
```diff
@@ -170,40 +178,62 @@ public void testClear() throws Exception {
-            Matchers.arrayContaining(4, 3, 2, 1)
-        );
+            new IsArrayContaining<>(
+                new AnyOf<>(
```
@umbrah `AllOf`?

llorllale commented on src/test/java/org/cactoos/collection/ReversedTest.java:
```diff
@@ -170,40 +178,62 @@ public void testClear() throws Exception {
-            Matchers.arrayContaining(5, 4, 3, 2, 1)
-        );
+            new IsArrayContaining<>(
+                new AnyOf<>(
```
@umbrah `AllOf`?

llorllale replied:
@Umbrah better yet, `HasValues`

llorllale commented on src/test/java/org/cactoos/collection/SkippedTest.java:
```diff
@@ -25,103 +25,126 @@
-            Matchers.contains(
-                "three",
-                "four"
+            new IsCollectionContaining<>(
```
@umbrah `HasValues`

llorllale commented:
@umbrah see above.

Basically I worry that the logic may not have translated correctly into the new refactored code. Also, take a look at the available matchers from `cactoos-matchers` - will probably make your life easier in most cases.

Umbrah commented:
@llorllale The logic was incorrect, I updated all usages of IsCollectionContaining<>(AnyOf<>(...)) with AllOf<>(IsCollectionContaining<>(...)). I cannot use HasValues and HasSize as we do not have it in last published version of cactoos-matchers. Please check.

llorllale commented:
@rultor merge

rultor commented:
> @rultor merge

@llorllale OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/18707-486480399)

rultor commented:
> @rultor merge

@llorllale Done! FYI, the full log is [here](http://www.rultor.com/t/18707-486480399) (took me 18min)

0crat commented:
Job was finished in 35 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

<!-- https://www.0crat.com/footprint/C63314D6Z/5c4e9f1a-adbd-4273-b459-1b60bcc3b5df, version: 0.46.6, hash: ${buildNumber} -->

0crat commented:
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

<!-- https://www.0crat.com/footprint/C63314D6Z/2ee16a42-7c8e-43cf-a345-40d9ce18f30f, version: 0.46.6, hash: ${buildNumber} -->

0crat commented:
Job [#1114](https://github.com/yegor256/cactoos/issues/1114) is not in the agenda of @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), can't inspect

<!-- https://www.0crat.com/footprint/C63314D6Z/59cfae25-27ab-4c4d-a288-8cfc45f6123f, version: 0.46.6, hash: ${buildNumber} -->

0crat commented:
The job [#1114](https://github.com/yegor256/cactoos/issues/1114) is now out of scope

<!-- https://www.0crat.com/footprint/C63314D6Z/cc927cde-e47f-4b07-a875-1d21943f0283, version: 0.46.6, hash: ${buildNumber} -->

0crat commented:
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

<!-- https://www.0crat.com/footprint/C63314D6Z/6511f35a-e2ff-4370-9ad4-0487755d4be1, version: 0.46.6, hash: ${buildNumber} -->

sereshqua commented:
@0crat quality good

0crat commented:
Order was finished, quality is "good": +25 point(s) just awarded to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov)

<!-- https://www.0crat.com/footprint/C63314D6Z/179c78ad-993e-44b5-bbb2-b3de3e192833, version: 0.46.6, hash: ${buildNumber} -->

0crat commented:
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

<!-- https://www.0crat.com/footprint/C63314D6Z/b4e40e89-5698-4102-9564-326472b70d33, version: 0.46.6, hash: ${buildNumber} -->
